mathics/builtin/__init__.py

```diff
@@ -37,6 +38,42 @@
 )
 
 
+def contributing_builtin_var(module, name) -> Optional[type]:
+    """
+    Returns getattr(module, name) if it is a Builtin that
+    should contribute with a definition.
+    Otherwise, returns None.
+    """
+    if name.startswith("_"):
+        return None
+
+    var = getattr(module, name)
+
+    if not (isinstance(var, type) and hasattr(var, "__module__")):
```

Comment:
[inspect.ismodule](https://docs.python.org/3/library/inspect.html#inspect.ismodule) is more idiomatic.

Reply:
What we want here is to check if var is a type, and if it was defined inside a module, not if it is a module

Reply:
Ok - you are correct again. A comment before this line indicating what you inform would be helpful. 

There is this nagging thing about this entire function that has all of these very specific conditions that feels wrong. 

Something like adding a class or module variable just feels simple and more direct in intent.

Reply:
@rocky, just to put this back in context. This PR started because in the formatting refactor, it was required a way to define base classes, subclasses from `Builtin` that 

>  *  To follow Python standards
>  *  To follow Mathics naming pattern
>  *  To avoid overpopulating mathics.builtin.base
>  *  To avoid adding unwanted base classes as a Definition.

There was a chunk of code that was repeated in several places and must be modified in order to get this feature. Then, I tried to avoid the repetition by defining a single function, and making the slightest change as possible to get it to work.  Then, the other change I did was just to split a long chain of `and` in parts. Then, maybe there is a better way to implement this function or even the whole Builtin listing mechanism, but I prefer to leave it to you or something that understands better how to do a better implementation.

Reply:
Yeah, I get that. All of this stuff is in `__init__.py` which is another thing that isn't a good idea or common practice either, and I haven't before said anything about that. 

There is just so much that is weird or charitably reflect Python programming circa over a decade ago, that it is impossible to fix in one got. 

In the PR, I currently have noted some of the weirdness here and added a FIXME.